greenmail-core/src/main/java/com/icegreen/greenmail/smtp/SmtpHandler.java

```diff
@@ -104,7 +104,7 @@ protected boolean commandLegalSize() {
             return false;
         }
 
-        if (currentLine.length() > 1000) {
```

Comment:
@marcelmay not sure if this is something to talk about.

Reply:
I believe this is the limit from RF821/[2821](https://datatracker.ietf.org/doc/html/rfc2821#section-4.5.3.1):
```
text line
      The maximum total length of a text line including the <CRLF> is
      1000 characters (not counting the leading dot duplicated for
      transparency).  This number may be increased by the use of SMTP
      Service Extensions.
```
It would be easy to extend of course, though many SMTP servers seem to adhere to the limit (and GreenMail would 'simulate' these).

Xoauth access token can be up to 4096, making the auth line > 4096, right ?

What about only checking length 1000 when the line beginning contains '<CODE> AUTHENTICATE XOAUTH2'?

Reply:
Did not found anything in regards to the "new" methods like XOAUTH2, but the 4096 make sense. Will change accordingly.